Sources/NIO/ServerSocket.swift

```diff
@@ -109,4 +109,17 @@
             return sock
         }
     }
+    
+    /// Close the socket.
+    ///
+    /// After the socket was closed all other methods will throw an `IOError` when called.
+    ///
+    /// - throws: An `IOError` if the operation failed.
+    override func close() throws {
+        let maybePathname = try? self.localAddress().pathname
+        try super.close()
+        if let socketPath = maybePathname {
+            try BaseSocket.cleanupSocket(unixDomainSocketPath: socketPath)
```

Comment:
With the above in mind, should we make `cleanupSocket` a method instead of a static?

Reply:
we could, but would probably still need a static version of that to clean up the file before binding (or pass the `cleanupExistingSocketFile` option through many steps)

Reply:
Fair enough.